### ISSUE TITLE
Parser fails if the first group by is a case statement

### ISSUE THREAD

**Issue by salonijuneja** [JSQLParser/JSqlParser#1308]
Hi, I am using parser to find underlying tables in a query.

**Parser failing on query:** 

SELECT (CASE WHEN 'USD' = 'USD' THEN "column1"
ELSE "column2"
END) AS "testColumn1", "testColumn2", "testColumn3", count(*)
FROM test_schema.table_name
group by  (CASE WHEN 'USD' = 'USD' THEN "column1" ELSE "column2" END), "testColumn2", "testColumn3"

**It works on query, though not entirely correct (able to format sql but removing commas in group by columns)**

SELECT (CASE WHEN 'USD' = 'USD' THEN "column1"
ELSE "column2"
END) AS "testColumn1", "testColumn2", "testColumn3", count(*)
FROM test_schema.table_name
group by "testColumn2", (CASE WHEN 'USD' = 'USD' THEN "column1" ELSE "column2" END), "testColumn3"

Both the SQLs are valid.
Tried out [here](http://jsqlformatter.manticore-projects.com/jsqlformatter/demo.html?args=-c%20LTAEDMEMEsBsFMAmoAuB7CaBOBbSLQBnFSLFAKAGUBRAGWoGEAVUACgYEEbQB1ACWoA5UAHIAqpQAiI0AF5RE6aCYDhAIgDGaWAFccAOwCMa8nW6bte-QCYTQyQEpQXUGpTxiDSweMAaV+6e3jZq-m4eKF66BgDMoaBaOvoorABUDuQAYgBKAPIAsqgRAPqEGgAW8HgAdCQARgjF+pA48OQA5lhoOgAOoHUAnqBsnNz8QgpSMvLiU8qqrlrRRmqgZtSLwbZrgo5hgZFb8eFBy3HkIBAwCMjomLj4oPD6iEA+)

**Comment by salonijuneja** [comment]
Hi @manticore-projects @wumpz I see that a ticket has been checked in and issue closed. But I wanted to confirm because this query is still failing in parser:

select * 
from dual 
group by (case when 1=1 then 'X' else 'Y' end), column1

Please let me know the resolution.
Tried [here](http://jsqlformatter.manticore-projects.com/jsqlformatter/demo.html?args=-c%20LTAEDMEMEsBsFMAmoAuB7CaBOBbSLQBnFSLFAKBAhgWXU13yJLMrCjiVQ3GzwOKkKoKh1rcG-ZkPKF4CAMYEAVKHLgsaHKEQBXSLDUBzTboAOoAEYBPUAAoFkOaADuAC3gA7UAEYAvD6oHt4A5AAaIaDyziEAmpFeiACUADSgCmiwujiePmzUnHQ8fEyJ+WJc9LyMBIlAA+)

Thanks,
Saloni

**Comment by manticore-projects** [comment]
Greetings,

what exactly have you tested please?
The fix as been committed and accepted into GIT Master Branch, but not released yet and (if I understand it correctly), also no Snapshot has been released yet.

You would need to pull from GIT and compile it yourself for using it.

The Online JSQLFormatter also not up-to-date but stuck with JSQLParser-4.1 although I plan to update it next week.

Best regards
Andreas

**Comment by wumpz** [comment]
Could you confirm, that the snapshot needs to be rebuild again? I thought it was done already.

**Comment by manticore-projects** [comment]
> Could you confirm, that the snapshot needs to be rebuild again? I thought it was done already.

After accepting PR #1318 all the builds should run normal again and Snapshots should reflect the latest Git.

**Comment by salonijuneja** [comment]
Hi @manticore-projects,

Is the JSQLFormatter also updated?

This query is still failing:

select * 
from dual 
group by (case when 1=1 then 'X' else 'Y' end), column1

Tired out [here](http://jsqlformatter.manticore-projects.com/jsqlformatter/demo.html?args=-c%20LTAEDMEMEsBsFMAmoAuB7CaBOBbSLQBnFSLFAKBAhgWXU13yJLPMPgQGMCAqUc8FjQ5QiAK6RY-AOZCxAB1AAjAJ6gAFJ0jtQAdwAW8AHagAjAF5TqQyYDkADVugOO2wE0nxxAEoANKE40WDEcI1NKMCg4JFQMcGw8Ai8I6mi6OISmLyA+)

**Comment by manticore-projects** [comment]
> Is the JSQLFormatter also updated?

Sorry, not yet. But I promise to pick it up soonest.

**Comment by salonijuneja** [comment]
Thanks

**Comment by manticore-projects** [comment]
Sorry for being late: http://jsqlformatter.manticore-projects.com/jsqlformatter/demo.html?args=-c%20MoUQMiDCAqAEsApaQIKgFDy9nuDqAEiAHKwDkAqsACJmwC85Vtmub780RpARAMYB7ADYBXALYA7AIw9WHDuFCx+w8RIBMs+bhLVYASlhplAFwCmAZxORVkmXJwAaU5eu2NWts57mrN0ZIAzJ64zjYiEiZIAFQG6ABiAEoA8gCysL4mAPoWfAAWZmIAhgB0JkUARkJmWRJFYmboAOIpFAAKsABCAJrYSKgY2kOwhCRMNHSMlBMOw9pcYyoB0iFzbIogyoLLmrNrunFD3pn+artHLn7uwegA3EA+

Everything has been updated to JSQLParser 4.3-SNAPSHOT.

Cheers!

**Comment by salonijuneja** [comment]
Thanks for all the efforts @manticore-projects 

@wumpz When do we plan to release it? I am asking because we use Maven central only in our firm :(

**Comment by manticore-projects** [comment]
> When do we plan to release it? I am asking because we use Maven central only in our firm :(

Only after the Starbucks Vouchers arrive :-D

More seriously: @Wumpz has released 4.3 last week with all the patches included.
You should be able to get it from the Maven Central already.

**Comment by wumpz** [comment]
I checked it: https://mvnrepository.com/artifact/com.github.jsqlparser/jsqlparser/4.2. This was the last version I released last week and it is already published. So can we close this or do you have a different problem?

**Comment by salonijuneja** [comment]
@wumpz , Yes I verified, failing queries are now being parsed by v4.2. The issue is already closed. Thanks!

Yes, there's firmwide repo parallel to maven central that contains all.

**Comment by wumpz** [comment]
@salonijuneja OK. Strange this restriction to only central. Many projects are published at there own repos. But since it is company restricion, who am I to argue? :)
